libc/src/__support/OSUtil/windows/io.cpp

```diff
@@ -0,0 +1,22 @@
+//===------------- Windows implementation of IO utils -----------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "io.h"
+#include "src/__support/macros/config.h"
+
+#define WIN32_LEAN_AND_MEAN
+#include <Windows.h>
```

Comment:
including the `Windows` headers here works, but it will cause problems if we ever want to do a fully standalone build. Given how windows is set up I'm fine with saying that windows is overlay only for now, but you should add a comment explaining that decision.

Reply:
It is actually hard to do the pure syscall approach with windows. They intentionally change the call ids.

Reply:
In that case it sounds like we should stick to overlay mode only.

Reply:
we probably should add some explicit error for Windows + full build in the cmake so that the errors people get on Windows won't be too confusing.

Reply:
Yeah, I think a working C23 library without POSIX would be the best but I am not sure whether there are other complications.

Reply:
> we probably should add some explicit error for Windows + full build in the cmake so that the errors people get on Windows won't be too confusing.

Updated in `libc/cmake/modules/LLVMLibCArchitectures.cmake` and made this a hard cmake configure error.